Assets/MixedRealityToolkit/Interfaces/InputSystem/IMixedRealityPointer.cs

```diff
@@ -116,5 +116,10 @@ public interface IMixedRealityPointer : IEqualityComparer
         /// Called after performing the scene query.
         /// </summary>
         void OnPostSceneQuery();
+
+        /// <summary>
+        /// Called during the scene query just before the current pointer target changes.
+        /// </summary>
+        void OnCurrentPointerTargetAboutToChange();
```

Comment:
>OnCurrentPointerTargetAboutToChange [](start = 13, length = 35)

Nit: I haven't really scene "aboutToChange" phrasing in APIs as a common thing (I basic search finds eclipse does this).

i.e. in keeping with other naming this could be OnPreCurrentPointerTargetChange()

This also allows us to add another function later called:

OnPostCurrentPointerTargetChange()

Instead of:

OnCurrentPointerTargetJustChanged()